component/plugincontroller/component.go

```diff
@@ -0,0 +1,34 @@
+package plugincontroller
+
+import (
+	"github.com/docker/docker/component"
+	"github.com/docker/docker/plugin"
+	"golang.org/x/net/context"
+)
+
+const name = "plugincontroller"
+
+// Set sets the plugin controller component
+func Set(s *plugin.Manager) (cancel func(), err error) {
+	return component.Register(name, s)
+}
```

Comment:
I had the same problem in  #26609. Separating the component interface and implementation? That way you can cast to the interface, but we can still have different implementations?

If we're never going to have different implementations, and we have to reference the implementation directly, why even have a registry? We could just split things out of the daemon struct and pass them around as references.